examples/05_scaling_non_linear_models.py

```diff
@@ -0,0 +1,537 @@
+#    take the first :math:`n` observations for the training set, the next
+#    :math:`m` observations for the test set and so on. This may not be the
+#    case for all datasets, so be careful before applying this code to your own
+#    !
+#
+#
+# Finally, the :code:`X` and :code:`y` are loaded.
+#
+#
+# .. topic:: Note: offsetting the test set
+#
+#    We create an offset to separate the training and the test set. The reason
+#    for this, is that the online procedures of this example will consume far
+#    more rows, but we still would like to compare accuracies with the same the
+#    same test set, and not change it each time. Therefore, we "reserve" the
+#    first 100000 rows for the training phase. The rest is made available to
+#    the test set.
+train_set_size = 5000
+test_set_size = 10000
+offset = 100000
+
+X_train, y_train = get_X_y(skiprows=1, names=columns_names,
+                           nrows=train_set_size)
+
+X_test, y_test = get_X_y(skiprows=offset, names=columns_names,
+                         nrows=test_set_size)
+
+###############################################################################
+# Evaluating time and sample complexity
+# -------------------------------------
+# Let's get an idea of model precision and performance depending on the number
+# of the samples used in the train set.  First, a small helper is created to
+# profile model fitting.
+import time
+def profile(func):
+    def profiled_func(*args, **kwargs):
+        t0 = time.perf_counter()
+        res = func(*args, **kwargs)
+        total_time = time.perf_counter() - t0
+        return total_time, res
+    return profiled_func
+
+###############################################################################
+# The |Pipeline| is then trained over different training set sizes. For this,
+# :code:`X_train` and :code:`y_train` get sliced into subsets of increasing
+# size, while :code:`X_test` and :code:`y_test` do not get changed when the
+# sample size varies.
+from sklearn.metrics import average_precision_score
+# define the different train set sizes on which to evaluate the model
+train_set_sizes = [train_set_size // 10, train_set_size // 3, train_set_size]
+
+def evaluate_model(model, X_train, y_train, X_test, y_test, train_set_sizes,
+                   one_hot_encoder):
+    train_times, test_scores = [], []
+
+    for n in train_set_sizes:
+        train_time, _ = profile(model.fit)(X_train[:n], y_train[:n])
+
+        y_pred = model.predict(X_test)
+
+        y_pred_onehot = one_hot_encoder.transform(y_pred.reshape(-1, 1))
+        y_test_onehot = one_hot_encoder.transform(y_test.reshape(-1, 1))
+
+        test_score = average_precision_score(y_test_onehot, y_pred_onehot)
+
+        train_times.append(train_time)
+        test_scores.append(test_score)
+    return train_times, test_scores
+
+train_times_svc, test_scores_svc = evaluate_model(model, X_train, y_train,
```

Comment:
Given that "evaluate_model" is used only once, I would not define a function, and I would simply inline the code, to make the example simpler.